platform/linux-generic/odp_fdserver.c

```diff
@@ -647,6 +647,32 @@ int _odp_fdserver_init_global(void)
 	}
 
 	if (server_pid == 0) { /*child */
+		sigset_t sigset;
+		struct sigaction action;
+
+		sigfillset(&sigset);
+		/* undefined if these are ignored, as per POSIX */
+		sigdelset(&sigset, SIGFPE);
+		sigdelset(&sigset, SIGILL);
+		sigdelset(&sigset, SIGSEGV);
+		/* can not be masked */
+		sigdelset(&sigset, SIGKILL);
+		sigdelset(&sigset, SIGSTOP);
+		/* these we want to handle */
+		sigdelset(&sigset, SIGTERM);
+		if (sigprocmask(SIG_SETMASK, &sigset, NULL) == -1) {
+			ODP_ERR("Could not mask terminal signals");
+			exit(1);
+		}
+
+		/* set default handlers for those signals we can handle */
+		action.sa_handler = SIG_DFL;
+		sigemptyset(&action.sa_mask);
+		action.sa_flags = 0;
+		sigaction(SIGKILL, &action, NULL);
+		sigaction(SIGSTOP, &action, NULL);
```

Comment:
You cannot set a signal handler for SIGKILL or SIGSTOP (see sigaction(2)).

Reply:
True, thanks!